File: sandbox-runner/patterns/sliding-window/typescript/shared.ts
export const TASK_QUEUE = "sliding-window-task-queue";
export const WORKFLOW_ID_PREFIX = "sliding-window";
export const WINDOW_SIZE = 3;
// Total records to process in the demo.
export const RECORD_IDS: string[] = Array.from({ length: 12 }, (_, i) => `record-${i}`);
export const COMPLETION_SIGNAL = "recordCompleted";

/** Input for SlidingWindowWorkflow. Bundled as a single object so all runs
 *  (including continues-as-new) share one consistent argument shape. */
export interface SlidingWindowInput {
  recordIds: string[];
  windowSize?: number;
  startIndex?: number;
  totalProcessed?: number;
  inFlight?: number;
}

File: sandbox-runner/patterns/sliding-window/python/shared.py
from dataclasses import dataclass, field

TASK_QUEUE = "sliding-window-task-queue"
WORKFLOW_ID_PREFIX = "sliding-window"
WINDOW_SIZE = 3
RECORD_IDS = [f"record-{i}" for i in range(12)]
COMPLETION_SIGNAL = "recordCompleted"


@dataclass
class SlidingWindowInput:
    """Input for SlidingWindowWorkflow. Bundled as a single object so all runs
    (including continues-as-new) share one consistent argument shape."""
    record_ids: list
    window_size: int = WINDOW_SIZE
    start_index: int = 0
    total_processed: int = 0
    in_flight: int = 0
